### ISSUE TITLE
clang-10 is using /usr/bin/ld which causes `unknown option: -platform_version` on OSX

### ISSUE THREAD

**Issue by s22chan** [conda-forge/clangdev-feedstock#102]
Issue:

clang-9 would use `$PREFIX/bin/ld`/$LD to link, which would insure that if you were running on an XCode 9.0 VM that things would work correctly.

clang-10 -v test.cc:
```
 "/usr/bin/ld" -demangle -lto_library /Users/distiller/.conda/envs/test/lib/libLTO.10.dylib -no_deduplicate -dynamic -arch x86_64 -platform_version macos 10.12.0 0.0.0 -syslibroot /Applications/Xcode-9.0.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -o a.out /var/folders/ms/xg67k5sn16xc7sdr_w3q45840000gn/T/test-09561e.o -lc++ -lSystem /Users/distiller/.conda/envs/test/lib/clang/10.0.0/lib/darwin/libclang_rt.osx.a
ld: unknown option: -platform_version
```

clang-9 -v test.cc:
```
 "/Users/distiller/.conda/envs/test/bin/ld" -demangle -lto_library /Users/distiller/.conda/envs/test/lib/libLTO.9.dylib -no_deduplicate -dynamic -arch x86_64 -macosx_version_min 10.12.0 -syslibroot /Applications/Xcode-9.0.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -o a.out /var/folders/ms/xg67k5sn16xc7sdr_w3q45840000gn/T/test-2f92e0.o -lc++ -lSystem /Users/distiller/.conda/envs/test/lib/clang/9.0.1/lib/darwin/libclang_rt.osx.a
```

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list
cctools_osx-64            949.0.1              h5ba7a2e_1    conda-forge
clang                     10.0.0          default_hf57f61e_3    conda-forge
clang-tools               10.0.0          default_hf57f61e_3    conda-forge
clang_osx-64              10.0.0               h05bbb7f_0    conda-forge
clangxx                   10.0.0          default_hf57f61e_3    conda-forge
compiler-rt               10.0.0               h47ead80_0    conda-forge
compiler-rt_osx-64        10.0.0               hbcc88fd_0    conda-forge
ld64_osx-64               530                  h3c32e8a_1    conda-forge
libclang-cpp10            10.0.0          default_hf57f61e_3    conda-forge
libcxx                    10.0.0               h1af66ff_2    conda-forge
libllvm10                 10.0.0               h7475705_1    conda-forge
llvm-tools                10.0.0               h1341992_1    conda-forge
llvmdev                   10.0.0               h7475705_1    conda-forge
tapi                      1000.10.8            ha1b3eb9_4    conda-forge
zlib                      1.2.11            h0b31af3_1006    conda-forge
```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info
     active environment : test
    active env location : /Users/distiller/.conda/envs/test
            shell level : 1
       user config file : /Users/distiller/.condarc
 populated config files : /Users/distiller/.condarc
          conda version : 4.8.3
    conda-build version : not installed
         python version : 3.7.7.final.0
       virtual packages : __osx=10.12.6
       base environment : /Users/distiller/.conda  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/osx-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://conda.anaconda.org/bioconda/osx-64
                          https://conda.anaconda.org/bioconda/noarch
          package cache : /Users/distiller/.conda/pkgs
       envs directories : /Users/distiller/.conda/envs
               platform : osx-64
             user-agent : conda/4.8.3 requests/2.23.0 CPython/3.7.7 Darwin/16.7.0 OSX/10.12.6
                UID:GID : 501:20
             netrc file : None
           offline mode : False
```
</details>

**Comment by isuruf** [comment]
Please use the `compilers` package or use `$CC` to compile.

**Comment by isuruf** [comment]
`unknown option: -platform_version` would be fixed with 10.0.1
